app/services/usps_in_person_proofing/transliterable_validator.rb

```diff
@@ -0,0 +1,55 @@
+module UspsInPersonProofing
+  # Validator that can be attached to a form or other model
+  # to verify that specific supported fields are transliterable
+  #
+  # == Example
+  #
+  #   validates_with UspsInPersonProofing::TransliterableValidator,
+  #     fields: [:first_name, :last_name]
+  #
+  class TransliterableValidator < ActiveModel::Validator
+    # Initialize the validator with the given fields configured
+    # for transliteration validation
+    #
+    # @param [Hash] options
+    # @option options [Array<Symbol>] fields Fields for which to validate transliterability
+    def initialize(options)
+      super
+      @fields = options[:fields]
+      if (@fields & SUPPORTED_FIELDS).size < @fields.size
+        # Catch incorrect usage of the validator
+        raise StandardError.new(
+          "Unsupported transliteration fields: #{(@fields - SUPPORTED_FIELDS).to_json}",
+        )
+      end
+    end
+
+    # Check if the configured values on the record are transliterable
+    #
+    # @param [ActiveModel::Validations] record
+    def validate(record)
+      return unless IdentityConfig.store.usps_ipp_transliteration_enabled
+      helper.validate(
+        @fields.index_with do |field|
+          (record.send(field) if record.respond_to?(field))
+        end,
+      )&.each do |key, value|
+        unless value.nil?
+          record.errors.add(
+            key, :nontransliterable_field, message: value,
+                                           pii: true
+          )
+        end
+      end
+    end
+
+    private
+
+    # Fields supported by this validator
+    SUPPORTED_FIELDS = UspsInPersonProofing::TransliterableValidatorHelper::SUPPORTED_FIELDS
```

Comment:
how come this and the helper are separate classes? when would we use one without the other? Can we combine them?

Reply:
The need to transliterate these fields is a requirement of the USPS API, not necessarily a specific form. Therefore it seemed appropriate to keep this split from a specific form validation mechanism.

Reply:
But we already have the `Transliterator` class that does the heavy lifting that's shared? Let's combine these now and we can split them up again if we find a use case that needs these to be separate?

Reply:
If I refactor away from this approach, then I would split the responsibilities between the validator and the forms that use the validator instead of putting this all directly into the validator.